pg_search/src/index/directory/mvcc.rs

```diff
@@ -97,6 +97,7 @@ pub struct MVCCDirectory {
     loaded_metas: OnceLock<Arc<tantivy::Result<IndexMeta>>>,
     all_entries: Arc<Mutex<HashMap<SegmentId, SegmentMetaEntry>>>,
     pin_cushion: Arc<Mutex<Option<PinCushion>>>,
+    total_segment_count: Arc<AtomicUsize>,
 }
 
 unsafe impl Send for MVCCDirectory {}
```

Comment:
Tangentially related, but: it looks like `PgSearchRelation` is the only thing which isn't actually `Send`/`Sync` in here, but by adding these impls, I think that we've turned off the checking for all of the fields... so when adding `total_segment_count`, I think that the compiler would have let you do it as just a `usize`, which would be incorrect.

Might be worth breaking out a tiny `PgSearchRelationWrapper` with the `Send`/`Sync`  impls.